src/main/java/com/atlassian/db/replica/internal/LockBasedThrottledCache.java

```diff
@@ -0,0 +1,35 @@
+package com.atlassian.db.replica.internal;
+
+import com.atlassian.db.replica.internal.util.ThreadSafe;
+import com.atlassian.db.replica.spi.SuppliedCache;
+
+import java.util.Optional;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Supplier;
+
+@ThreadSafe
+public final class LockBasedThrottledCache<T> implements SuppliedCache<T> {
```

Comment:
Should it be hidden by `compatibleWithPreviousVerions` 🤔

Reply:
We use `compatibleWithPreviousVerions` for the cases when you can't opt-out from the change via API.

Reply:
Ach, yes, this cache is a part of API. OK.